@here/harp-mapview/lib/Statistics.ts

```diff
@@ -1074,6 +1070,18 @@ export class PerformanceStatistics {
                 this.currentFrame.setValue("memory.jsHeapSizeLimit", memory.jsHeapSizeLimit);
             }
         }
+    }
+
+    /**
+     * Stores the current frame events into the array of events and clears all values.
+     *
+     * @returns {boolean} Returns `false` if the maximum number of storable frames has been reached.
+     * @memberof PerformanceStatistics
+     */
+    storeAndClearFrameInfo(): boolean {
+        if (this.m_frameEvents.length >= this.maxNumFrames) {
+            return false;
+        }
```

Comment:
I guess it would be much better to store frames info in the round buffer, this way we never loose last frames information... Maybe this is a cause why we got some unpredictable results in some cases.
BTW: I know, it's not your code :).

Reply:
true. for now I assume the buffer is big enough.

Reply:
this seems to be able to store 1000 frames, at 60 fps, that is roughly 15 seconds worth of data, for a flyover test, that may not be enough?

Reply:
I think the flyover test is fetching the statistics after every frame, so the buffer will not overflow but I will double check

Reply:
@nzjony : Actually your suspicion was correct. By default we store 1000 frames but flyover renders 5000 without fetching statistics in between. What is also interesting is, that m_frameEvents is actually a ring buffer but we just stop putting statistics into it.

Reply:
After looking again into our code and due to hints from @CraigBerlin I found out that the `PerformanceStatistics` are not really following the singleton pattern, i.e. `PerformanceStatistics.instance` will be recreated whenever a new `PerformanceStatistics` object is instantiated.